File: benchmarks/cli/pipe_10k.go
package main

import (
	"fmt"
	"strings"
	"time"
)

func main() {
	lines := make([]string, 10000)
	for i := range lines {
		lines[i] = fmt.Sprintf("line %d: the quick brown fox", i)
	}
	start := time.Now()
	for i := range lines {
		lines[i] = strings.ToUpper(lines[i])
	}
	fmt.Println(time.Since(start).Microseconds())
}

File: benchmarks/cli/pipe_10k.rs
use std::time::Instant;
fn main() {
    let lines: Vec<String> = (0..10000)
        .map(|i| format!("line {}: the quick brown fox", i))
        .collect();
    let start = Instant::now();
    let _upper: Vec<String> = lines.iter().map(|l| l.to_uppercase()).collect();
    println!("{}", start.elapsed().as_micros());
}
